### DISCUSSION TITLE
A question for all of Bolt's contributors

### DISCUSSION THREAD

**Discussion by markkasaboski** [bolt/core#3518] [General]
Hello everyone,

I came across Bolt recently as I was looking for a Symfony based CMS for a project that I'm starting. It was obvious at first that momentum had stopped on this repo, but at the time I wasn't sure why. I started playing with Bolt anyway. It was until yesterday that I learned of Bob's passing. 

So my question to everyone is: Is anyone interesting in keeping Bolt going? Has anyone here forked and kept it moving? I did a quick search, but didn't find anything. I'd be interesting in helping it to move forward. Is anyone else interested? I'm sure Bob would want it to carry on.

Thanks.

**Comment by saling-star** [comment]
I think we all, people using Bolt, are waiting for keeping it alive ! 
It becomes urgent to adapt Bolt to Symfony last release since mainenance of version 5.4 will end shortly.
For Bolt 5.2 :
Package sensio/framework-extra-bundle is abandoned, you should avoid using it. Use Symfony instead.
Package symfony/web-server-bundle is abandoned, you should avoid using it. No replacement was suggested.
Package tightenco/collect is abandoned, you should avoid using it. Use illuminate/collections instead.

Does anyone know how to update the deprecated packages ? Wil it be enought for Symfony 7 ?

**Comment by cervelas** [comment]
Hello Everyone !

As everyone here, i love Bolt for it's simplicity and it's power. I use it on a few website, it is very reliable. (i've 2 installs still on bolt 3.7!)
I think this project deserve at least a 5.2 maintenance release. 

The big issue is that the PR's tests requirements are not pass anymore due thoses deprecations.

I'm willing to help, i'm not a symfony expert, but i can create a PR to update the Core regarding all thoses composer warnings:

Package sensio/framework-extra-bundle is abandoned, you should avoid using it. Use Symfony instead.
Package symfony/web-server-bundle is abandoned, you should avoid using it. No replacement was suggested.
Package tightenco/collect is abandoned, you should avoid using it. Use illuminate/collections instead.
Package webmozart/path-util is abandoned, you should avoid using it. Use symfony/filesystem instead.

This will obviously require refactoring of the code using thoses libs.

The next steps would be to updates the others projects as needed (e.g. bolt/article, bolt/assets, etc...), then update to Symfony 7, then make a 5.2 release.

Who is willing to help with that ? i'm throwing a bottle, maybe @bobvandevijver @nestordedios @Sander-Hub  ?

Bonus question, who can update the PR settings ? can anyone have admin access to this project on github ?

Cheers,
C

**Reply by bobvandevijver** [reply]
While I would love helping with this project (I currently have 4 projects running on it, 5.2.0-rc.1), I'm afraid I currently do not have the time to commit to that 😞

**Reply by avdavidw** [reply]
I've loved Bolt and have used it a lot. I've played with a few new installs and tried to update some of the outdated components and it just seemed to open a can of worms into more and more that needed updates and code changes. 

My assessment, at least for me, would be that time would be better spent creating something new than trying to fix Bolt. If there were a few people willing to split it up, I'd be willing to take a piece, though.

**Comment by celtic34fr** [comment]
Hello everyone

I would like to submit an idea to the community.
Bolt CMS seems to be waiting, stay there. The latest version published is V 5.2.0-rc.1 (22/12/2023) / 5.1.24 (19/02/2023) and many issues remain (203).

I appreciate Bolt CMS for its modularity and the fact that even inexperienced programmers on Symfony can quickly modify it by bringing their need.

Bob den Otter is dead, which hurts me because his work for Bolt CMS is fabulous. It would be even sadder if it does not progress anymore and falls into oblivion like many projects on GitHub.

In order not to let Bolt CMS become a new dead project, I propose to constitute with the current contributors and those who want to become one, a plan in 3 phases :
_ first, stabilize and produce version 5.2 taking into account only the problems that have been opened to date,
* migrate Bolt CMS from Symfony 5.* to 6.* because Symfony version 5.4 is coming to the end of its life,
* migrate Bolt CMS to Symfony 7.*, this less in the emergency (version 6.4 is maintained until 30 Nov 2027)

This plan will make live the work of Bob. it will also be necessary in parallel to make a reorganization of the site listing and allowing to obtain extensions and themes: it is necessary to distinguish according to the version of Bolt CMS compatible with the extension or theme, and to simplify the proposal of extension, of theme, via a form.

I await your opinion on this idea of continuation of Bolt CMS.

Gilbert ARMENGAUD
celtic34fr@gmail.com
Beziers - Herault - France

**Reply by bobvandevijver** [reply]
Actually, Symfony 5.4 will be supported (looking at security fixes only) a lot longer than 6.4 (February 2029 versus November 2027). So keeping this project of 5.4 for now is not the biggest issue, and I would skip 6.4 completely. 7.4 will be released in November this year, and is supported till November 2029 (so longer than 5.4).

I've been using Bolt 5.2.0-rc.1 for quite some time without any issue, so 5.2.0 can be tagged directly imo.

**Reply by cervelas** [reply]
Agree, the biggest issue for me is the CI/PR devops that are not passing anymore. i also noted that the docker compose are based on php 7.3. I'm ok to get my hands dirty to fix thoses.

**Reply by celtic34fr** [reply]
@bobvandevijver happy to read your answer.  But your proposal is too heavy to be realistic, I think the transition to a version under Symfony 6.* will allow a smooth transition.

I work locally, on the transition to version 6.0.* to already take a first step towards a version 6.4.*. There will already be enough difficulties to negotiate and solve in the transition to a major version higher, the work seems to me more enormous to want to move to the second version higher as you propose. I think you have to climb small mountains first before you want to climb Everest.

**Reply by bobvandevijver** [reply]
You're probably forgetting that when you support 6.4 with no deprecations, 7.x will directly work without adjustments. So, yes, 6.4 will be the intermediate to help solve all deprecations, but I see no reason to maintain and support a bolt version just for 6.x.

**Reply by celtic34fr** [reply]
that's my goal

**Comment by saling-star** [comment]
Hello  Bolt-Lovers,
I use this CMS (4 sites) since 3-4 years and therefor very much concerned by a lack of maintenance.
My needs are :
- multi-sites on a single app and data-base (symfony multi-kernel was too difficult, a tinkering in templates more easy)
- compatibility with symfony last version (security,  updates).

We all want to continue with Bolt but...
Some of you want to prolong Bob's project as it was, but it seems a big effort (too few people in the community, no more manager...),
Others are septic (discovering "a can of worms" !!! in bolt-core) and prefer  restart from the beginning a new one.

My proposal is in the middle, I think that Bolt Bob's project was genial, perhaps too genial and therefor too much difficult to maintain. 
I am working on Symfony-Demo (a very simple demo-blog compatible with Symfony 7.2) with the aim to reuse the existing data-base of Bolt (Content, Field and FieldTranslation -Taxonomy ?). At the end the public interface of the CMS will follow  Symfony at it's last version and the private interface stay **temporarily** on Sf5.4 but only accessible by admin  on a private server (so far from Internet hackers).

Sorry if it is completely out of subject but I am novice in object-programming, I can't understand Git-Hub functioning (fork, branches, PR...). Sorry for my French-English-charabia...
Saling Paris France

**Comment by cervelas** [comment]
Thanks everyone for your comments so far,

So i believe we can start something here, at least making a working 5.2 release.

There is a proposal for a roadmap to the 5.2 release, the idea is to have the release process working, and some dependency update as bonus points.

1. Analyse then Repair / update Github actions, there is some deprecated actions and others stuff, this is obviously number one for me.
2. Update Dockerfile and docker compose to a more php version (current 7.3 !)
3. Update composer file and dependencies to support ^8.2 version, deprecation warnings are acceptable for now.

Bonus points:

- PHP 8.3 or 8.4 support
- Bump to Symfony 7.x
- Resolve some Issues ! 

Who is motivated to help ?

**Reply by celtic34fr** [reply]
@cervelas, your proposal is interesting, but in the first place, it will already be necessary to set up a project governance board to submit your proposals and decide on the roadmap to follow.

The question then remains: Who is willing to take time to allow the project to continue ?.

**Reply by bobvandevijver** [reply]
Bolt currently requires `"php": ">=7.2.9"`, so "support" for 8.3 and 8.4 is already there. I'm running it with 8.3 ever since 8.3 has been released, no issues there.

I would be open for PR review and the occasional development when needed for my usage. But, rights to the repository and possibly the Github group would be required. Maybe @Sander-Hub can help me with that?

**Reply by celtic34fr** [reply]
Will it not be necessary finaly to make a fork of all the bolt/* repositories in bolt6/* to signify the passage of version of Symfony and have total control over all the processes of Github ?

**Reply by bobvandevijver** [reply]
Not if the rights to the Github organisation can be passed over, which would have absolute preferences over yet another fork.

**Reply by evertalbers** [reply]
I will be happy to act as test driver/crash test dummy and maybe update parts of the manual when required. But I am afraid that I don't know much about Php or Symfony.

**Reply by positron48** [reply]
I have a blog on Bolt 5, and Symfony with PHP 8 is my main working stack. You can count on my help.
I suppose the first step is to decide on a new location, as access to the current repositories has been lost, and continuing development here is no longer an option?

**Reply by cervelas** [reply]
@celtic34fr Agree that this is not the place to manage project :-), I noticed that the wiki is editable, maybe we can create a project/roadmap there for now ? what do you think ?

I agree with @bobvandevijver that it will be much more better to reclaim rights on the github organisation.
Apparently we have to contact github to reclaim the organisation as described here https://docs.github.com/en/site-policy/other-site-policies/github-deceased-user-policy. 
Anyone here have already done that ? We need someone that have contributed (or better someone who know Bob relatives) to this repo contact github support... but who ?

Beside all that, i'm ok to work with some guys on this nice project !

**Reply by bobvandevijver** [reply]
That is why I mentioned @Sander-Hub, he seems to have some rights for at least the boltcms.io domain (see https://github.com/bolt/core/issues/3522#issuecomment-2308494292)

**Reply by Sander-Hub** [reply]
Thanks everyone for taking the conversation here to the github repo (the Slack deletes messages after 3 months). It is the more appropriate place for it.

Normally I've only been a member of the github organisation. But last week, when Loic, Gilbert and Clem and BobvdV started hatching somewhat of a plan and had converations on Slack about that and asked for potential volunteers I contacted Ivo, the last remaining member with owner rights (who is retired from the project). And I have acquired owner rights to the github organisation from him. So kudos to Ivo for helping out with that.

Now I guess I should make the following folks members of the organisation:

- @bobvandevijver aka BobvdV
- @cervelas aka Clem
- @celtic34fr aka Gilbert

am I missing anyone?

I am willing to help out. Althought I have very little expierence with developing in Symfony, mainly small things extending Bolt. But I know a thing or two and also about using Bolt.

About BobvdV remark. Yes, the Bolt sites are still on Two Kings infrastructure, which is still active. At the moment this is graciously continued but this will not last indefinitely obviously. Last June/July 2024 the boltcms.io was briefly put on hold status due to reasons (Bob being the owner and deceased). Those issues got resolved. Then we still have the old bolt.cm domain, which I used last summer as a stopgap measure to ensure the docs were reachable for people during that period.

The search in the docs do not work for Bolt 5.2 due to fact that the Algolia widget used is no longer supported. I've been looking for an alternative on and off.

This is a bit the gist of it since Bob's passing.

And shoutout to @evertalbers , nice to see familiair names.

-Sander

**Reply by cervelas** [reply]
This is very good news, shout out to you @Sander-Hub and also to Ivo ! 

On my side it's ok to be added as member of the organisation, with pleasure.

Concerning the hosting stuff, What if we want to update the boltcms.io site ? Someone at Two Kings is ok to update it ? (@Sander-Hub you are working there ?), or is there some external access available from Two King infra or some kind of git repo to push on to update the site ?
i also noticed that some sub-domain are not working anymore, like [slack.boltcms.io](https://slack.bolt.cm/) or [people.bolt.cm](https://people.boltcms.io/), we have to assess the need for thoses domains.

The search in the docs will be added in the "broken" list or something like that.

As a side not i discovered the Bob Memoriam, and it really touched me, what a guy ! (https://twokings.nl/bobfeitjes)

**Reply by Sander-Hub** [reply]
> or is there some external access available from Two King infra or some kind of git repo to push on to update the site ?

Not for a third party at the moment.

The Bolt sites are Bolt sites ;-) , meaning any content updates are done via the CMS. Actual Bolt updates or changes are done by depoying said changes or branch with deployer (PHP deployement tool; deployer.org).

Decoupling the sites from Two Kings infrastucture would be a logical step.

people.boltcms.io is a subdomain of a Bolt community member. I suspect that member shut down the site. The site is not directly under our control (we can remove DNS records if needed though).

**Reply by simplecreatif** [reply]
Like a few others, I do not have the knowledge or skill in Symfony, backend work or extending the core of Bolt as I am but a mere frontend dev. I do however, LOVE Bolt and have built about 30 sites with it. I would be happy to contribute money or testing time, or anything that I could to help get a 5.2 release up and running.

**Reply by bobvandevijver** [reply]
I've got limited access to the Github organisation now (limited because I cannot access the setting, we'll see how far I can come without that) and will look whether I can create the 5.2 release sometime soon. 

Is the Slack workspace still being used? If so, can somebody create an invite so I can join?

**Reply by cervelas** [reply]
@bobvandevijver I think the rights are ok, i've created the [5.2.0 draft release](https://github.com/bolt/core/releases/edit/untagged-2360c055df18604dd5f8), it's empty for now.

**Reply by evertalbers** [reply]
> Is the Slack workspace still being used? If so, can somebody create an invite so I can join?

Yes it is. Me/somebody else would need your email address for an invite, if I see it correctly.

**Reply by bobvandevijver** [reply]
You can use `bob [at] drenso [dot] nl` for the invite

**Reply by Sander-Hub** [reply]
FYI. https://slack.bolt.cm is working again (I had no idea this was a thing by the way).

**Comment by celtic34fr** [comment]
Bonjour à l'équipe.

Pour les échanges sur le plan dév. ne serait-il pas plus pratique de créer un serveur Discord ? Facilité pour les échanges de message, mais possibilité de travail avec partage d'écran et web-conférence azu besoin. Je vous soumets cela pour étude.
Il serait bon de contacter @AntiFTW (https://boltcms.slack.com/team/U01F2MYUUKW) qui travaille sur la question de la monté de version de Symfony, nous avons échangé sur le sujet et il semblez avoir bien avancé dessus.

Hello to the team.

For discussions on the dev plan, wouldn't it be more practical to create a Discord server? Easy for message exchanges, but possibility of working with screen sharing and web-conference if needed. I submit this to you for study.
It would be interesting to contact @AntiFTW (https://boltcms.slack.com/team/U01F2MYUUKW) who is already working on the issue of Symfony's version upgrade, we have discussed the subject and it seems to have made good progress on it.

Gilbert ARMENGAUD
celtic34fr
Béziers - Hérault - France

**Reply by macintoshplus** [reply]
Hello @celtic34fr,

As for the GitHub organization, Slack has been the community's means of communication for a while. Now that it's working again, we might as well use it.

For the migration work to Symfony 6, this is very good news. However, I fear that there are too many changes in a single PR. The best thing is to separate the changes into smaller PRs.


Bonjour @celtic34fr,

Comme pour l'organisation GitHub, le slack est le moyen de communication de la communauté depuis un moment. Maintenant qu'il fonctionne à nouveau autant l'utiliser.

Pour le travail de migration vers Symfony 6, c'est une très bonne nouvelle. Je crin cependant qu'il n'y ai trop de modification dans une seule PR. Le mieux étant de séparer les évolutions en PR plus petite.

Jean-Baptiste.

**Comment by markkasaboski** [comment]
Oh wow! I wasn't getting notifications from Github for a while. I just came back to this now and seeing all of this. I've joined the Slack and I'll see how I can help out.

**Comment by ashceo** [comment]
Hi all, first of all thank you for continue contributing and making Bolt CMS alive! Further development should be focused not on symfony update itself which is easy job, currently outdated doctrine/orm used with annotations stopping us moving forward, so in my opinion we should update doctrine to newer version and use attributes, this job was mostly done by @antiftw so we can take some of his code and apply to current code base and finish everything, the we can update symfony/api-platform etc. Also i have done updating my fork to symfony 6.4.x with some gotchas without updating doctrine so its not hard to do at the end, to have full upgrade doctrine is first to do.

Also the Slack itself is okay but probably creating a Discord server may be the way for the future.

**Reply by macintoshplus** [reply]
The next major release of Bolt will include updates for the ORM, ApiPlatform, and Symfony 7.4 or Symfony 6.4.

I've already created PRs to update some elements. But this requires a lot of review time.
To merge a PR, it must be validated by someone other than its author.

Currently, only @bobvandevijver (whom I thank) does this in his free time.

If more contributors could review PRs to make Bolt better, that would be great.

---------------------------------------------------------------------------

La prochaine version majeur de Bolt sera à jour pour l'ORM, ApiPlatform et Symfony 7.4 ou Symfony 6.4.

J'ai déjà réalisé des PR pour mettre à jour certaine chose. Mais cela demande beaucoup de temps pour les relectures.
Pour fusionner une PR elle doit être valider par quelqu'un d'autre que l'auteur de la PR.

Actuellement, seul @bobvandevijver  (que je remercie) le fait sur son rare temps libre.

Si plus de contributeur pouvait relire les PR pour aider à faire évoluer Bolt ce serait génial.
